http/src/main/java/com/hazelcast/jet/contrib/http/impl/HttpListenerSinkContext.java

```diff
@@ -66,8 +66,9 @@ public HttpListenerSinkContext(
             @Nonnull FunctionEx<T, String> toStringFn
     ) {
         this.accumulateLimit = accumulateLimit;
-        this.messageBuffer = accumulateLimit > 0 ? new ArrayList<>(accumulateLimit) : null;
-        this.sinkHttpHandler = sinkType == WEBSOCKET ? new WebSocketSinkHttpHandler() : new ServerSentSinkHttpHandler();
+        this.messageBuffer = accumulateLimit > 0 ? new ConcurrentLinkedQueue<>() : null;
```

Comment:
Code style: we could make the `messageBuffer` not-null always and eliminate some of the ifs below. In case of zero limit we can use a non-concurrent queue to save on volatile reads.

Reply:
👍

Reply:
I thought you could you `Queue<String>` for the `messageBuffer` and here do this:
```java
this.messageBuffer = accumulateLimit > 0 ? new ConcurrentLinkedQueue<>() : new LinkedList<>();
```
Without any ad-hoc classes...

Reply:
I take it back, I noticed you also address other concerns here.